sycl/include/sycl/detail/group_sort_impl.hpp

```diff
@@ -200,10 +189,9 @@ void bubble_sort(Iter first, const size_t begin, const size_t end,
   }
 }
 
-template <typename Group, typename Iter, typename Compare>
-void merge_sort(Group group, Iter first, const size_t n, Compare comp,
-                std::byte *scratch) {
-  using T = typename GetValueType<Iter>::type;
+template <typename Group, typename T, typename Compare>
+void merge_sort(Group group, T *first, const size_t n, Compare comp,
+                T *scratch) {
```

Comment:
Why this change?

Reply:
Now we don't pass the pointer to the scratch as `std::byte *` but do placement new converting it to T*, so we pass it as T* now.

Reply:
I meant `Iter` -> `T*`, not the `std::byte` -> `T` change.

Reply:
Indeed, that's unnecessary limitation, reverted.